hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/execution/TestBoundedInMemoryExecutorInSpark.java

```diff
@@ -152,6 +165,144 @@ protected Integer getResult() {
     } finally {
       if (executor != null) {
         executor.shutdownNow();
+        executor.awaitTermination();
+      }
+    }
+  }
+
+  @Test
+  public void testExecutorTermination() throws ExecutionException, InterruptedException {
```

Comment:
Appreciate your effort in putting up this test, but i usually suggest to avoid checking in non-deterministic tests -- non-deterministic tests is paving the way for flakiness of them given that we don't control tests execution environment (in CI and where not)

Let's instead either try to rewrite it as deterministic test (both positive, negative) by controlling the execution with `CountDownLatch`, `CyclicBarrier` OR keep just the positive case (which has to pass and should not fail)

Reply:
Do think about this. This test cannot achieve the goal to find out if a code change can reproduce this problem. Make this UT more simple which just validate termination and awaiting works well.